dockerfile

```diff
@@ -1,23 +1,78 @@
-FROM ubuntu:20.04
-
-# install python resources
-RUN apt-get update && apt-get install -qqy build-essential python3.8 python3-pip
+FROM --platform=linux/amd64 ubuntu:20.04 AS napari
 
 # below env var required to install libglib2.0-0 non-interactively
-ENV TZ America/Los_Angeles
-ENV DEBIAN_FRONTEND noninteractive
-
-# install graphical libraries used by qt and vispy
-RUN apt-get install -qqy mesa-utils libgl1-mesa-glx  libglib2.0-0
-RUN apt-get install -qqy libfontconfig1 libxrender1 libdbus-1-3 libxkbcommon-x11-0 libxi6
-RUN apt-get install -qqy libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0
-RUN apt-get install -qqy libxcb-xinerama0 libxcb-xinput0 libxcb-xfixes0 libxcb-shape0
+ENV TZ=America/Los_Angeles
+ARG DEBIAN_FRONTEND=noninteractive
 
-# install napari release version
-RUN pip3 install napari[all]
+# install python resources + graphical libraries used by qt and vispy
+RUN apt-get update && \
+    apt-get install -qqy  \
+        build-essential \
+        python3.8 \
+        python3-pip \
+        git \
+        mesa-utils \
+        libgl1-mesa-glx \
+        libglib2.0-0 \
+        libfontconfig1 \
+        libxrender1 \
+        libdbus-1-3 \
+        libxkbcommon-x11-0 \
+        libxi6 \
+        libxcb-icccm4 \
+        libxcb-image0 \
+        libxcb-keysyms1 \
+        libxcb-randr0 \
+        libxcb-render-util0 \
+        libxcb-xinerama0 \
+        libxcb-xinput0 \
+        libxcb-xfixes0 \
+        libxcb-shape0
 
-# install scikit image for examples
-RUN pip3 install scikit-image
+# install napari release version + scikit-image (for examples)
+RUN pip3 install napari[all] scikit-image
 COPY examples /tmp/examples
 
 ENTRYPOINT ["python3", "-m", "napari"]
+
+#########################################################
+# Extend napari with a preconfigured Xpra server target #
+#########################################################
+
+FROM napari AS napari-xpra
+
+# Install Xpra and dependencies
+RUN apt-get install -y wget gnupg2 apt-transport-https && \
+    wget -O - https://xpra.org/gpg.asc | apt-key add - && \
+    echo "deb https://xpra.org/ focal main" > /etc/apt/sources.list.d/xpra.list
+
+RUN apt-get update && \
+    apt-get install -yqq \
+        xpra \
+        xvfb \
+        xterm \
+        sshfs && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
could we move this to the first `apt-get` call. It will reduce the docker file size.

Reply:
Unfortunately we can't. We are trying to keep things separate (`napari` target is a different image from `napari-xpra`) and we need to add the ppa first anyway. Final size will be the same, the only difference being cacheability. With this split, `napari` users can "upgrade" to `napari-xpra` by just downloading the new layers, instead of re-downloading a bigger layer.